### PR TITLE
Extend preprocessor LLVM version checks to support LLVM 4.x

### PR DESCRIPTION
**Pull request by shepmaster**
This doesn't actually do anything for LLVM 4.x yet, but sets the stage.

### CODE REVIEW

**Comment by sanxiyn** [comment]
cc #36295

File: src/rustllvm/rustllvm.h [hunk 1]
```diff
@@ -45,7 +45,16 @@
 #include "llvm-c/ExecutionEngine.h"
 #include "llvm-c/Object.h"
 
-#if LLVM_VERSION_MINOR >= 7
+#define LLVM_VERSION_GT_OR_EQ(major, minor) \
+  LLVM_VERSION_MAJOR > (major) || LLVM_VERSION_MAJOR == (major) && LLVM_VERSION_MINOR >= (minor)
```
Comment (alexcrichton) [inline src/rustllvm/rustllvm.h]: Could this entire clause also be wrapped in parens? (yay C macros)

same below as well

File: src/rustllvm/rustllvm.h [hunk 2]
```diff
@@ -45,7 +45,16 @@
 #include "llvm-c/ExecutionEngine.h"
 #include "llvm-c/Object.h"
 
-#if LLVM_VERSION_MINOR >= 7
+#define LLVM_VERSION_GT_OR_EQ(major, minor) \
```
Comment (alexcrichton) [inline src/rustllvm/rustllvm.h]: Perhaps this could be `LLVM_VERSION_GE`?

Similar to `ParialOrd::ge` naming at least

**Review by alexcrichton — COMMENTED** [review]
Thanks @shepmaster! Looks great to me.

File: src/rustllvm/rustllvm.h (hunk 2 above)
Comment (shepmaster) [inline src/rustllvm/rustllvm.h]: Done!

**Pull request merged**